core/include/detray/navigation/navigation_config.hpp

```diff
@@ -22,22 +26,39 @@ enum class trust_level {
 };
 
 /// Navigation configuration
-template <typename scalar_t>
```

Comment:
less templating? 😱

Reply:
I know :D

Reply:
First I thought this might limit you a bit but for options it is definitely a good idea to have just a single interface. I guess `float` over `double` is a performance decision? But also it might not really have an impact to use `double` over `float` in most cases

Reply:
Actually, I am following what traccc did in its config. Using ```double``` in the config could lead to implicitly converting some calculation to double precision in a single precision build, which is indeed not good on the majority of GPUs performancewise. So, the CI is set to fail on implicit conversions, which for this PR would result in much more static casts back to single precision than now.

Reply:
Since some GPUs fundamentally cannot do FP64 operations, if we have to choose one scalar type for the config objects, we indeed need to choose `float`. As was said already, for configuration parameters 32-bits should be enough. We can then do double precision calculations using these single precision configuration values where needed. :thinking: